### DISCUSSION TITLE
xmake 生成 compile_commands 不为系统 packages 生成相关参数

### DISCUSSION THREAD

**Discussion by yangyagami** [xmake-io/xmake#6021] [Q&A]
![raylib info](https://github.com/user-attachments/assets/eb5fb3d7-9953-4e33-8fd5-882d1bdf268e)
![compile_commands.json](https://github.com/user-attachments/assets/b37c11f6-b2ed-4342-aa4f-47957b0b7a2d)

**Comment by waruqi** [comment]
我没 nixos ，调不了，直接调用 find_package 和 add_requires 里面的 find ，状态差异挺多的，没法比较。

你可以自己调下相关代码

**Comment by star-hengxing** [comment]
建议直接复制字符串贴上来而不是截图。

---

xmake 应该没对 nix 包管理做支持，那么应该是走 pkgconfig 找的。据我所知，好像有一些库的 .pc 会塞一些不必要的参数

**Comment by yangyagami** [comment]
是我自己的问题，我没有手动添加 add_packages
